File: Pizza-Ordering/src/app/Models/Product.ts
export class Product {
  productId: number | undefined
  name: string | undefined
  image: string | undefined
  description: string | undefined
  price: number | undefined
  productCategoryId: number | undefined
  sizeId: number | undefined
  crustId: number | undefined
  isActive: boolean | undefined
  quantity: number | undefined
  size: Size | undefined
  crust: Crust | undefined
  productCategory: Category | undefined
  pizzaToppings: Array<Topping> | undefined;
  pizzaSelectedToppingNames: string | undefined;
  isCustomized: boolean = false;
  priceAfterCustomization: number | undefined;
  total: number | undefined;
}

export class Category {
  categoryId: number | undefined;
  name: string | undefined;
  isActive: boolean | undefined;
}


export class Topping {
  toppingId: number | undefined
  name: string | undefined
  price: number | undefined
  isActive: boolean | undefined
}

export class Size {
  sizeId: number | undefined
  name: string | undefined
  price: number | undefined
  isActive: boolean | undefined
}

export class Crust {
  crustId: number | undefined
  name: string | undefined
  price: number | undefined
  isActive: boolean | undefined
}


export class ProductWrapper {
  productCategory: Category | undefined
  products: Array<Product> | undefined
}

export class Menu {
  products: Array<ProductWrapper> | undefined;

}

export class Order {
  customerNumber: string | undefined
  sessionId: string | undefined
  TotalPrice: number | undefined
  orderDateTime: Date | undefined
  deliverDateTime: Date | undefined
  status: number | undefined
  products: Array<Product> | undefined
}

File: Pizzaria.Entities/DataModels/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Pizzaria.Entities.DataModels
{
    public class Product : IDbEntity
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Image { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int ProductCategoryId { get; set; }
        public int? SizeId { get; set; }
        public int? CrustId { get; set; }
        public bool IsActive { get; set; }
        public int Quantity { get; set; }

        public virtual Size Size { get; set; }
        public virtual Crust Crust { get; set; }
        public virtual Category ProductCategory { get; set; }

        public ICollection<Topping> PizzaToppings { get; set; }
    }
}
